src/cli/circom.rs

```diff
@@ -0,0 +1,84 @@
+use std::{
+    fs::{self, File},
+    io::copy,
+    path::PathBuf,
+    process::Command,
+};
+
+use anyhow::Result;
+use sha2::{Digest, Sha256};
+
+use super::paths::non_wasm::{circom_binary, circom_gadgets};
+
+/// Attempts to verify the checksum and version of the circom binary.
+/// If the binary does not exist, it is downloaded.
+/// TODO
```

Comment:
This may help:
https://gist.github.com/huitseeker/ff00e7ca970b2c12f62e2d58542e9a1c
As most of our code base isn't async, `smol::block_on` may help as well:
https://github.com/smol-rs/smol

Reply:
Thank you!